rocky/reports/views/generate_report.py

```diff
@@ -234,23 +239,70 @@ class SetupScanGenerateReportView(
     current_step = 3
 
     def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        del request.session["report_name"]  # delete session
+        del request.session["reference_date"]  # delete session
+        if not self.report_has_required_plugins() or self.plugins_enabled():
+            return redirect(self.get_next())
         if not self.selected_report_types:
             messages.error(self.request, _("Select at least one report type to proceed."))
             return redirect(
                 reverse("generate_report_select_report_types", kwargs=self.get_kwargs()) + get_selection(self.request)
             )
-        if not self.report_has_required_plugins() or self.plugins_enabled():
-            report_ooi = self.save_report()
-            return redirect(
-                reverse("view_report", kwargs={"organization_code": self.organization.code})
-                + "?"
-                + urlencode({"report_id": report_ooi.reference})
-            )
         if not self.plugins:
             return redirect(self.get_previous())
         return super().get(request, *args, **kwargs)
 
 
+class ExportSetupGenerateReportView(
+    GenerateReportStepsMixin, BreadcrumbsGenerateReportView, ReportPluginView, TemplateView
+):
+    """
+    Shows the export setup page where users can set their export preferences.
+    """
+
+    template_name = "generate_report/export_setup.html"
+    breadcrumbs_step = 6
+    current_step = 4
+
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        if not self.selected_report_types:
+            messages.error(request, _("Select at least one report type to proceed."))
+            return redirect(self.get_previous())
+        if "report_name" not in request.session:
+            request.session["report_name"] = self.set_report_name()
+
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        new_report_name = self.request.POST.get("report-name")
+        new_reference_date = str(self.request.POST.get("reference-date"))
+        request.session["report_name"] = new_report_name
+        request.session["reference_date"] = new_reference_date
+        return redirect(self.get_current())
+
+    def set_report_name(self):
```

Comment:
Ideally, this should be an isolated function with arguments `oois_pk` and `report_types` that should be passed. This would improve readability and maintainability by reducing code tangling